bundles/org.openhab.binding.roku/src/main/java/org/openhab/binding/roku/internal/communication/RokuCommunicator.java

```diff
@@ -283,8 +285,12 @@ public List<Channel> getTvChannelList() throws RokuHttpException {
      */
     private String getCommand(String url) throws RokuHttpException {
         try {
-            return httpClient.newRequest(url).method(HttpMethod.GET).timeout(REQUEST_TIMEOUT, TimeUnit.MILLISECONDS)
-                    .send().getContentAsString();
+            final String response = httpClient.newRequest(url).method(HttpMethod.GET)
+                    .timeout(REQUEST_TIMEOUT, TimeUnit.MILLISECONDS).send().getContentAsString();
+            if (response != null && response.contains(LIMITED_MODE_RESPONSE)) {
+                throw new RokuHttpException(LIMITED_MODE_RESPONSE);
```

Comment:
Minor: You could create a subclass of `RokuHttpException`, e.g. `RokuLimitedModeException`, so you could handle it directly in `RokuHandler` rather than comparing the message as you do in `isLimitedModeResponse`. The constant `LIMITED_MODE_RESPONSE` could then also be made private in this class.